src/evo/simplifiedmns.cpp

```diff
@@ -178,6 +180,66 @@ bool CSimplifiedMNListDiff::BuildQuorumsDiff(const CBlockIndex* baseBlockIndex,
             newQuorums.emplace_back(*qc);
         }
     }
+
+    return true;
+}
+
+bool CSimplifiedMNListDiff::BuildQuorumChainlockInfo(const CBlockIndex* blockIndex)
+{
+    // Group quorums (indexes corresponding to entries of newQuorums) per CBlockIndex containing the expected CL signature in CbTx.
+    // We want to avoid to load CbTx now, as more than one quorum will target the same block: hence we want to load CbTxs once per block (heavy operation).
+    std::multimap<const CBlockIndex*, uint16_t>  workBaseBlockIndexMap;
+
+    uint16_t idx = 0;
+    for (const auto& e : newQuorums) {
+        auto opt_params = llmq::GetLLMQParams(e.llmqType);
+        assert(opt_params.has_value());
+        Consensus::LLMQParams llmq_params = opt_params.value();
+        auto q = llmq::quorumManager->GetQuorum(e.llmqType, e.quorumHash);
+        const CBlockIndex* pWorkBaseBlockIndex = nullptr;
+        if (llmq_params.useRotation) {
+            // In case of rotation, all rotated quorums rely on the CL sig expected in the cycleBlock (the block of the first DKG) - 8
+            pWorkBaseBlockIndex = blockIndex->GetAncestor(q->m_quorum_base_block_index->nHeight - q->qc->quorumIndex - 8);
+        }
+        else {
+            // In case of non-rotation, quorums rely on the CL sig expected in the block of the DKG - 8
+            pWorkBaseBlockIndex = blockIndex->GetAncestor(q->m_quorum_base_block_index->nHeight - 8);
+        }
+
+        workBaseBlockIndexMap.insert(std::make_pair(pWorkBaseBlockIndex, idx));
+        idx++;
+    }
+
+    auto it = workBaseBlockIndexMap.begin();
+    while (it != workBaseBlockIndexMap.end()) {
+        // Iterate the std::multimap once per key (CBlockIndex containing the expected CL signature in CbTx)
+        const CBlockIndex* pWorkBaseBlockIndex = it->first;
+        auto cbcl = GetNonNullCoinbaseChainlock(pWorkBaseBlockIndex);
+        CBLSSignature sig;
+        if (cbcl.has_value()) {
+            sig = cbcl.value().first;
+        }
+        // Get the range of indexes (values) for the current key and merge them into a single std::set
+        auto range = workBaseBlockIndexMap.equal_range(pWorkBaseBlockIndex);
```

Comment:
I think better to have a `const` modifier here for range - the `range` is not changing anyhow